src/libraries/Microsoft.PowerFx.Interpreter/Environment/OptionSet.cs

```diff
@@ -46,6 +50,22 @@ public OptionSet(string name, IDictionary<string, string> options)
         /// </summary>
         public ImmutableDictionary<DName, DName> Options { get; }
 
+        /// <summary>
+        /// Formula Type corresponding to this option set.
+        /// Use in record/table contexts to define the type of fields using this option set.
+        /// </summary>
+        public FormulaType FormulaType { get; }
+
+        public OptionSetValue GetValue(DName fieldName)
+        {
+            if (!Options.ContainsKey(fieldName)) 
+            {
+                throw new InvalidOperationException($"{fieldName.Value} is not a member of option set {EntityName.Value}");
```

Comment:
Is this an internal assert or part of user code validation?

Reply:
Moved to TryGet pattern